packages/next/shared/lib/router/router.ts

```diff
@@ -1829,7 +1829,9 @@ export default class Router implements BaseRouter {
     }
 
     const preflight = await this._getPreflightData({
-      preflightHref: options.as,
+      preflightHref: hasBasePath(options.as)
```

Comment:
Can we move this to the [prefetch call](https://github.com/vercel/next.js/blob/9fe0fda5b47f690a7d1b40169b0a5a9c58d9c53d/packages/next/shared/lib/router/router.ts#L1700) instead of checking if the basePath was already added?

Reply:
Thanks for the suggestion! Now I'm looking it again, indeed that makes more sense.
In the last commit https://github.com/vercel/next.js/pull/31101/commits/3b923bf0a527afae3aec3de4ad4e7c875de14c8d, I updated the code as you suggested.